syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -303,20 +339,109 @@ def __truediv__(self, shares: dict, divisor):
         for location, pointer in shares.items():
             divided_shares[location] = pointer / divisor
 
-        return AdditiveSharingTensor(
-            field=self.field, crypto_provider=self.crypto_provider
-        ).set_shares(divided_shares)
+        return divided_shares
 
     @overloaded.method
     def mod(self, shares: dict, modulus: int):
         assert isinstance(modulus, int)
+
         moded_shares = {}
         for location, pointer in shares.items():
             moded_shares[location] = pointer % modulus
 
-        return AdditiveSharingTensor(
-            field=self.field, crypto_provider=self.crypto_provider
-        ).set_shares(moded_shares)
+        return moded_shares
 
     def __mod__(self, *args, **kwargs):
         return self.mod(*args, **kwargs)
+
+    ## SECTION SNN
+
+    def relu(self):
+        return securenn.relu(self)
+
+    def positive(self):
+        # self >= 0
+        return securenn.relu_deriv(self)
+
+    def __gt__(self, other):
+        r = self - other - 1
+        print(r)
+        return r.positive()
+
+    def __ge__(self, other):
+        return (self - other).positive()
+
+    def __lt__(self, other):
+        return (other - self - 1).positive()
+
+    def __le__(self, other):
+        return (other - self).positive()
+
+    def __eq__(self, other):
+        diff = self - other
+        diff2 = diff * diff
+        negdiff2 = diff2 * -1
+        return negdiff2.positive()
+
+    def eq(self, other):
```

Comment:
why is this different from __eq__

Reply:
It's a redirection from tensor.eq to tensor.__eq__, which is hooked correctly, those are different. It is important because for torch tensor I redirected tensor.__eq__ to tensor.eq as mentioned in #2030